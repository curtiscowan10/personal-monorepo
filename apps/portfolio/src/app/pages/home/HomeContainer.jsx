import React from 'react';
import {
  Brush as DesignIcon,
  Code as CodeIcon,
  Speed as SpeedIcon,
} from '@mui/icons-material';
import Home from './Home';

const HomeContainer = () => {
  const skills = [
    'React',
    'JavaScript',
    'Node.js',
    'Material-UI',
    'SCSS',
    'Vite',
    'NX',
    'Git',
    'Responsive Design',
  ];

  const features = [
    {
      icon: <CodeIcon className="feature-icon-code" />,
      title: 'Clean Code',
      description:
        'Writing maintainable, scalable, and well-documented code following best practices.',
    },
    {
      icon: <DesignIcon className="feature-icon-design" />,
      title: 'Modern Design',
      description:
        'Creating beautiful, intuitive user interfaces with attention to detail and user experience.',
    },
    {
      icon: <SpeedIcon className="feature-icon-speed" />,
      title: 'Performance',
      description:
        'Optimizing applications for speed, accessibility, and cross-browser compatibility.',
    },
  ];

  return <Home skills={skills} features={features} />;
};

export default HomeContainer;
