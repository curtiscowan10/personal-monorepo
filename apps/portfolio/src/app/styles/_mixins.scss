// _mixins.scss - Reusable SCSS Mixins

@use 'variables' as *;
@use 'sass:color';
@use 'sass:math';

// Media Query Mixins
@mixin mobile {
  @media (max-width: #{$breakpoint-md - 1px}) {
    @content;
  }
}

@mixin tablet {
  @media (min-width: #{$breakpoint-md}) and (max-width: #{$breakpoint-lg - 1px}) {
    @content;
  }
}

@mixin desktop {
  @media (min-width: #{$breakpoint-lg}) {
    @content;
  }
}

@mixin mobile-only {
  @media (max-width: #{$breakpoint-sm - 1px}) {
    @content;
  }
}

// Layout Mixins
@mixin flex-center {
  display: flex;
  align-items: center;
  justify-content: center;
}

@mixin flex-between {
  display: flex;
  align-items: center;
  justify-content: space-between;
}

@mixin flex-column {
  display: flex;
  flex-direction: column;
}

@mixin flex-column-center {
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
}

// Container Mixins
@mixin container {
  max-width: 1200px;
  margin: 0 auto;
  padding: 0 $spacing-sm;

  @include tablet {
    padding: 0 $spacing-lg;
  }

  @include desktop {
    padding: 0 $spacing-xl;
  }
}

@mixin section-spacing {
  padding: $spacing-xl 0;

  @include mobile {
    padding: $spacing-lg 0;
  }
}

// Card Mixins
@mixin card {
  background: $color-white;
  border-radius: $border-radius-md;
  box-shadow: $shadow-base;
  overflow: hidden;
}

@mixin card-hover {
  transition: all $transition-base;

  &:hover {
    box-shadow: $shadow-lg;
    transform: translateY(-2px);
  }
}

@mixin card-content {
  padding: $spacing-lg;

  @include mobile {
    padding: $spacing-md;
  }
}

// Button Mixins
@mixin button-base {
  display: inline-flex;
  align-items: center;
  justify-content: center;
  padding: $spacing-sm $spacing-lg;
  border: none;
  border-radius: $border-radius-md;
  font-weight: $font-weight-medium;
  text-decoration: none;
  cursor: pointer;
  transition: all $transition-base;

  &:disabled {
    opacity: 0.6;
    cursor: not-allowed;
  }
}

@mixin button-primary {
  @include button-base;
  background-color: $color-secondary;
  color: $color-white;

  &:hover:not(:disabled) {
    background-color: color.adjust($color-secondary, $lightness: -10%);
  }
}

@mixin button-secondary {
  @include button-base;
  background-color: transparent;
  color: $color-secondary;
  border: 2px solid $color-secondary;

  &:hover:not(:disabled) {
    background-color: $color-secondary;
    color: $color-white;
  }
}

@mixin button-outline {
  @include button-base;
  background-color: transparent;
  border: 2px solid currentColor;

  &:hover:not(:disabled) {
    background-color: currentColor;
    color: $color-white;
  }
}

// Text Mixins
@mixin text-center {
  text-align: center;
}

@mixin text-truncate {
  overflow: hidden;
  text-overflow: ellipsis;
  white-space: nowrap;
}

@mixin text-gradient {
  background: linear-gradient(135deg, $color-secondary, $color-accent);
  -webkit-background-clip: text;
  -webkit-text-fill-color: transparent;
  background-clip: text;
}

// Hero Section Mixins
@mixin hero-section {
  // All CSS declarations first
  background: linear-gradient(135deg, $color-gray-50 0%, $color-gray-100 100%);
  border-radius: $border-radius-md;
  margin-bottom: $spacing-xl;
  text-align: center;

  // Then includes and nested rules
  @include section-spacing;

  @include mobile {
    padding: $spacing-xl 0;
  }
}

@mixin hero-avatar {
  width: 120px;
  height: 120px;
  margin: 0 auto $spacing-md auto;
  background-color: $color-accent;
  font-size: $font-size-3xl;
  border-radius: $border-radius-full;
  @include flex-center;
}

// Form Mixins
@mixin form-input {
  width: 100%;
  padding: $spacing-sm;
  border: 1px solid $color-gray-300;
  border-radius: $border-radius-sm;
  font-size: $font-size-base;
  transition: border-color $transition-base;

  &:focus {
    outline: none;
    border-color: $color-secondary;
    box-shadow: 0 0 0 3px rgba($color-secondary, 0.1);
  }
}

@mixin form-label {
  display: block;
  margin-bottom: $spacing-xs;
  font-weight: $font-weight-medium;
  color: $color-gray-700;
}

// Animation Mixins
@mixin fade-in {
  opacity: 0;
  animation: fadeIn $transition-base forwards;
}

@keyframes fadeIn {
  to {
    opacity: 1;
  }
}

@mixin slide-up {
  transform: translateY(20px);
  opacity: 0;
  animation: slideUp $transition-base forwards;
}

@keyframes slideUp {
  to {
    transform: translateY(0);
    opacity: 1;
  }
}

// Social Button Mixins
@mixin social-button($color) {
  @include button-outline;
  color: $color;
  border-color: $color;

  &:hover:not(:disabled) {
    background-color: $color;
    color: $color-white;
  }
}

// Grid Mixins
@mixin grid($columns: 1, $gap: $spacing-lg) {
  display: grid;
  grid-template-columns: repeat($columns, 1fr);
  gap: $gap;
}

@mixin responsive-grid($mobile: 1, $tablet: 2, $desktop: 3, $gap: $spacing-lg) {
  @include grid($mobile, $gap);

  @include tablet {
    grid-template-columns: repeat($tablet, 1fr);
  }

  @include desktop {
    grid-template-columns: repeat($desktop, 1fr);
  }
}
