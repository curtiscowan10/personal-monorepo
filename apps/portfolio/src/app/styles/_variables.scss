// _variables.scss - SCSS Constants and Variables

// Colors
$color-primary: #0f172a; // slate primary
$color-secondary: #3b82f6; // tech blue
$color-accent: #1976d2; // material blue
$color-white: #ffffff;
$color-black: #000000;
$color-gray-50: #fafafa;
$color-gray-100: #f5f5f5;
$color-gray-200: #eeeeee;
$color-gray-300: #e0e0e0;
$color-gray-700: #616161;
$color-gray-900: #212121;

// Social Media Colors
$color-linkedin: #0077b5;
$color-github: #333333;
$color-twitter: #1da1f2;

// Spacing
$spacing-xs: 0.5rem; // 8px
$spacing-sm: 1rem; // 16px
$spacing-md: 1.5rem; // 24px
$spacing-lg: 2rem; // 32px
$spacing-xl: 3rem; // 48px
$spacing-2xl: 4rem; // 64px
$spacing-3xl: 6rem; // 96px

// Font Sizes
$font-size-xs: 0.75rem; // 12px
$font-size-sm: 0.875rem; // 14px
$font-size-base: 1rem; // 16px
$font-size-lg: 1.125rem; // 18px
$font-size-xl: 1.25rem; // 20px
$font-size-2xl: 1.5rem; // 24px
$font-size-3xl: 1.875rem; // 30px
$font-size-4xl: 2.25rem; // 36px

// Font Weights
$font-weight-normal: 400;
$font-weight-medium: 500;
$font-weight-semibold: 600;
$font-weight-bold: 700;

// Border Radius
$border-radius-sm: 4px;
$border-radius-md: 8px;
$border-radius-lg: 12px;
$border-radius-xl: 16px;
$border-radius-full: 50%;

// Breakpoints
$breakpoint-xs: 480px;
$breakpoint-sm: 640px;
$breakpoint-md: 768px;
$breakpoint-lg: 1024px;
$breakpoint-xl: 1280px;
$breakpoint-2xl: 1536px;

// Z-index
$z-index-dropdown: 1000;
$z-index-sticky: 1020;
$z-index-fixed: 1030;
$z-index-modal-backdrop: 1040;
$z-index-modal: 1050;
$z-index-popover: 1060;
$z-index-tooltip: 1070;

// Transitions
$transition-fast: 0.15s ease;
$transition-base: 0.3s ease;
$transition-slow: 0.5s ease;

// Shadows
$shadow-sm: 0 1px 2px 0 rgba(0, 0, 0, 0.05);
$shadow-base: 0 1px 3px 0 rgba(0, 0, 0, 0.1), 0 1px 2px 0 rgba(0, 0, 0, 0.06);
$shadow-md: 0 4px 6px -1px rgba(0, 0, 0, 0.1),
  0 2px 4px -1px rgba(0, 0, 0, 0.06);
$shadow-lg: 0 10px 15px -3px rgba(0, 0, 0, 0.1),
  0 4px 6px -2px rgba(0, 0, 0, 0.05);
$shadow-xl: 0 20px 25px -5px rgba(0, 0, 0, 0.1),
  0 10px 10px -5px rgba(0, 0, 0, 0.04);
