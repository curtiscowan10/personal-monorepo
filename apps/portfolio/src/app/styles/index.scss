// index.scss - Main styles entry point

// Import variables and mixins first
@use 'variables' as *;
@use 'mixins' as *;

// Global styles
* {
  box-sizing: border-box;
}

body {
  font-family: Inter, -apple-system, BlinkMacSystemFont, 'Segoe UI', Roboto,
    sans-serif;
  margin: 0;
  padding: 0;
  line-height: 1.6;
  color: $color-primary;
  background-color: $color-white;
}

// Global utility classes following BEM-like naming
.u-text-center {
  @include text-center;
}

.u-text-truncate {
  @include text-truncate;
}

.u-text-gradient {
  @include text-gradient;
}

.u-fade-in {
  @include fade-in;
}

.u-slide-up {
  @include slide-up;
}

// Global component classes
.c-container {
  @include container;
}

.c-section {
  @include section-spacing;
}

.c-card {
  @include card;

  &--hover {
    @include card-hover;
  }

  &__content {
    @include card-content;
  }
}

.c-button {
  @include button-base;

  &--primary {
    @include button-primary;
  }

  &--secondary {
    @include button-secondary;
  }

  &--outline {
    @include button-outline;
  }
}

.c-form {
  &__input {
    @include form-input;
  }

  &__label {
    @include form-label;
  }
}

.c-grid {
  &--responsive {
    @include responsive-grid;
  }

  &--2-col {
    @include responsive-grid(1, 2, 2);
  }

  &--3-col {
    @include responsive-grid(1, 2, 3);
  }
}
