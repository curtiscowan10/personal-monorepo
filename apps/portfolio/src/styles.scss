/* Main styles entry point - imports the complete styling system */
@use 'app/styles/index';
