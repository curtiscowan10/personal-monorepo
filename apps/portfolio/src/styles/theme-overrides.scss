// Material UI Theme Overrides
// This file contains SCSS overrides for Material UI components

// Button overrides
.MuiButton-root {
  &.custom-primary {
    background: linear-gradient(45deg, #1976d2 30%, #42a5f5 90%);
    border: 0;
    border-radius: 12px;
    box-shadow: 0 3px 5px 2px rgba(25, 118, 210, 0.3);
    color: white;
    height: 48px;
    padding: 0 30px;
    transition: all 0.3s ease;

    &:hover {
      transform: translateY(-2px);
      box-shadow: 0 6px 10px 4px rgba(25, 118, 210, 0.3);
    }
  }

  &.custom-secondary {
    background: linear-gradient(45deg, #dc004e 30%, #ff5983 90%);
    border: 0;
    border-radius: 12px;
    box-shadow: 0 3px 5px 2px rgba(220, 0, 78, 0.3);
    color: white;
    height: 48px;
    padding: 0 30px;
    transition: all 0.3s ease;

    &:hover {
      transform: translateY(-2px);
      box-shadow: 0 6px 10px 4px rgba(220, 0, 78, 0.3);
    }
  }
}

// Card overrides
.MuiCard-root {
  &.custom-card {
    background: rgba(255, 255, 255, 0.95);
    backdrop-filter: blur(10px);
    border: 1px solid rgba(255, 255, 255, 0.2);
    border-radius: 16px;
    box-shadow: 0 8px 32px rgba(0, 0, 0, 0.1);
    transition: all 0.3s ease;

    &:hover {
      transform: translateY(-4px);
      box-shadow: 0 12px 40px rgba(0, 0, 0, 0.15);
    }
  }
}

// AppBar overrides
.MuiAppBar-root {
  &.custom-appbar {
    background: rgba(255, 255, 255, 0.9);
    backdrop-filter: blur(10px);
    border-bottom: 1px solid rgba(0, 0, 0, 0.1);
    box-shadow: 0 2px 20px rgba(0, 0, 0, 0.1);
    color: #212121;
  }
}

// Typography overrides
.MuiTypography-root {
  &.custom-title {
    background: linear-gradient(45deg, #1976d2, #dc004e);
    background-clip: text;
    -webkit-background-clip: text;
    -webkit-text-fill-color: transparent;
    font-weight: 700;
    text-align: center;
  }

  &.custom-subtitle {
    color: #757575;
    font-weight: 400;
    text-align: center;
    margin-bottom: 2rem;
  }
}

// Container overrides
.custom-container {
  padding: 2rem;
  min-height: 100vh;
  background: linear-gradient(135deg, #fafafa 0%, #f5f5f5 100%);
}

// Navigation overrides
.custom-nav {
  display: flex;
  justify-content: center;
  align-items: center;
  gap: 2rem;
  margin: 2rem 0;

  .nav-link {
    text-decoration: none;
    color: #1976d2;
    font-weight: 500;
    padding: 0.5rem 1rem;
    border-radius: 8px;
    transition: all 0.3s ease;

    &:hover {
      background-color: rgba(25, 118, 210, 0.1);
      transform: translateY(-2px);
    }

    &.active {
      background-color: #1976d2;
      color: white;
    }
  }
}

// Responsive design
@media (max-width: 768px) {
  .custom-container {
    padding: 1rem;
  }

  .custom-nav {
    flex-direction: column;
    gap: 1rem;
  }

  .MuiButton-root {
    &.custom-primary,
    &.custom-secondary {
      width: 100%;
      margin-bottom: 1rem;
    }
  }
}
