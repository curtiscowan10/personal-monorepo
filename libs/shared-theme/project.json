{
  "name": "shared-theme",
  "$schema": "../../node_modules/nx/schemas/project-schema.json",
  "sourceRoot": "libs/shared-theme/src",
  "projectType": "library",
  "tags": []
}
