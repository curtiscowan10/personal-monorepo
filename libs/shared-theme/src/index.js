export { theme } from './lib/theme';
export { default } from './lib/theme';
