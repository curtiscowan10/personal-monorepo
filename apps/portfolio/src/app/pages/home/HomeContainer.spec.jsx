import React from 'react';
import { render, screen } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import { ThemeProvider } from '@mui/material/styles';
import { theme } from '@personal-monorepo/shared-theme';
import HomeContainer from './HomeContainer';

const renderWithProviders = (component) => {
  return render(
    <BrowserRouter>
      <ThemeProvider theme={theme}>{component}</ThemeProvider>
    </BrowserRouter>
  );
};

describe('HomeContainer', () => {
  it('should render successfully', () => {
    renderWithProviders(<HomeContainer />);
    expect(screen.getByText('Welcome to My Portfolio')).toBeTruthy();
  });

  it('should render hero section with correct content', () => {
    renderWithProviders(<HomeContainer />);

    expect(screen.getByText('Welcome to My Portfolio')).toBeTruthy();
    expect(
      screen.getByText(/Full-Stack Developer passionate about creating/)
    ).toBeTruthy();
    expect(screen.getByTestId('home-page')).toBeTruthy(); // Avatar
  });

  it('should render all technology skills as chips', () => {
    renderWithProviders(<HomeContainer />);

    const expectedSkills = [
      'React',
      'JavaScript',
      'Node.js',
      'Material-UI',
      'SCSS',
      'Vite',
      'NX',
      'Git',
      'Responsive Design',
    ];

    expect(screen.getByText('Technologies I Work With')).toBeTruthy();

    expectedSkills.forEach((skill) => {
      expect(screen.getByText(skill)).toBeTruthy();
    });
  });

  it('should render features section with all three features', () => {
    renderWithProviders(<HomeContainer />);

    expect(screen.getByText('What I Bring to the Table')).toBeTruthy();

    expect(screen.getByText('Clean Code')).toBeTruthy();
    expect(
      screen.getByText(
        'Writing maintainable, scalable, and well-documented code following best practices.'
      )
    ).toBeTruthy();

    expect(screen.getByText('Modern Design')).toBeTruthy();
    expect(
      screen.getByText(
        'Creating beautiful, intuitive user interfaces with attention to detail and user experience.'
      )
    ).toBeTruthy();

    expect(screen.getByText('Performance')).toBeTruthy();
    expect(
      screen.getByText(
        'Optimizing applications for speed, accessibility, and cross-browser compatibility.'
      )
    ).toBeTruthy();
  });

  it('should render navigation buttons with correct links', () => {
    renderWithProviders(<HomeContainer />);

    const viewWorkButton = screen.getByRole('link', { name: /view my work/i });
    const getInTouchButton = screen.getByRole('link', {
      name: /get in touch/i,
    });
    const startConversationButton = screen.getByRole('link', {
      name: /start a conversation/i,
    });

    expect(viewWorkButton.getAttribute('href')).toBe('/projects');
    expect(getInTouchButton.getAttribute('href')).toBe('/contact');
    expect(startConversationButton.getAttribute('href')).toBe('/contact');
  });

  it('should render call to action section', () => {
    renderWithProviders(<HomeContainer />);

    expect(screen.getByText('Ready to Start Your Next Project?')).toBeTruthy();
    expect(
      screen.getByText("Let's work together to bring your ideas to life")
    ).toBeTruthy();
  });

  it('should display feature icons correctly', () => {
    const { container } = renderWithProviders(<HomeContainer />);

    // Should have feature icons with correct CSS classes
    expect(container.querySelector('.feature-icon-code')).toBeTruthy();
    expect(container.querySelector('.feature-icon-design')).toBeTruthy();
    expect(container.querySelector('.feature-icon-speed')).toBeTruthy();
  });

  it('should have correct CSS classes applied', () => {
    const { container } = renderWithProviders(<HomeContainer />);

    expect(container.querySelector('.home')).toBeTruthy();
    expect(container.querySelector('.home__hero')).toBeTruthy();
    expect(container.querySelector('.home__features')).toBeTruthy();
    expect(container.querySelector('.home__cta')).toBeTruthy();
  });

  it('should match snapshot', () => {
    const { container } = renderWithProviders(<HomeContainer />);
    expect(container.firstChild).toMatchSnapshot();
  });
});
