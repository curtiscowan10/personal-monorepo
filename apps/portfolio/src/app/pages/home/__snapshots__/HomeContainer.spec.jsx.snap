// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`HomeContainer > should match snapshot 1`] = `
<div
  class="MuiContainer-root MuiContainer-maxWidthLg home css-5c1adp-MuiContainer-root"
>
  <div
    class="home__hero MuiBox-root css-0"
  >
    <div
      class="MuiAvatar-root MuiAvatar-circular MuiAvatar-colorDefault home__hero-avatar css-1qcqc3p-MuiAvatar-root"
    >
      P
    </div>
    <h1
      class="MuiTypography-root MuiTypography-h2 MuiTypography-gutterBottom custom-title css-jp450z-MuiTypography-root"
    >
      Welcome to My Portfolio
    </h1>
    <h5
      class="MuiTypography-root MuiTypography-h5 home__hero-subtitle css-rgpxvr-MuiTypography-root"
    >
      Full-Stack Developer passionate about creating exceptional web experiences with modern technologies and clean, efficient code.
    </h5>
    <div
      class="MuiStack-root home__hero-buttons css-14w1sp1-MuiStack-root"
    >
      <a
        class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeLarge MuiButton-containedSizeLarge MuiButton-colorPrimary MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeLarge MuiButton-containedSizeLarge MuiButton-colorPrimary custom-primary css-dnximp-MuiButtonBase-root-MuiButton-root"
        href="/projects"
        tabindex="0"
      >
        View My Work
        <span
          class="MuiButton-icon MuiButton-endIcon MuiButton-iconSizeLarge css-1u6cks-MuiButton-endIcon"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1umw9bq-MuiSvgIcon-root"
            data-testid="ArrowForwardIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="m12 4-1.41 1.41L16.17 11H4v2h12.17l-5.58 5.59L12 20l8-8z"
            />
          </svg>
        </span>
      </a>
      <a
        class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeLarge MuiButton-outlinedSizeLarge MuiButton-colorPrimary MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeLarge MuiButton-outlinedSizeLarge MuiButton-colorPrimary css-3qwqpo-MuiButtonBase-root-MuiButton-root"
        href="/contact"
        tabindex="0"
      >
        Get In Touch
      </a>
    </div>
    <div
      class="home__skills MuiBox-root css-0"
    >
      <h6
        class="MuiTypography-root MuiTypography-h6 MuiTypography-gutterBottom css-f6ezis-MuiTypography-root"
      >
        Technologies I Work With
      </h6>
      <div
        class="MuiStack-root home__skills-container css-miphnx-MuiStack-root"
      >
        <div
          class="MuiChip-root MuiChip-outlined MuiChip-sizeMedium MuiChip-colorPrimary MuiChip-outlinedPrimary home__skills-chip css-mo0pjq-MuiChip-root"
        >
          <span
            class="MuiChip-label MuiChip-labelMedium css-16cgrcw-MuiChip-label"
          >
            React
          </span>
        </div>
        <div
          class="MuiChip-root MuiChip-outlined MuiChip-sizeMedium MuiChip-colorPrimary MuiChip-outlinedPrimary home__skills-chip css-mo0pjq-MuiChip-root"
        >
          <span
            class="MuiChip-label MuiChip-labelMedium css-16cgrcw-MuiChip-label"
          >
            JavaScript
          </span>
        </div>
        <div
          class="MuiChip-root MuiChip-outlined MuiChip-sizeMedium MuiChip-colorPrimary MuiChip-outlinedPrimary home__skills-chip css-mo0pjq-MuiChip-root"
        >
          <span
            class="MuiChip-label MuiChip-labelMedium css-16cgrcw-MuiChip-label"
          >
            Node.js
          </span>
        </div>
        <div
          class="MuiChip-root MuiChip-outlined MuiChip-sizeMedium MuiChip-colorPrimary MuiChip-outlinedPrimary home__skills-chip css-mo0pjq-MuiChip-root"
        >
          <span
            class="MuiChip-label MuiChip-labelMedium css-16cgrcw-MuiChip-label"
          >
            Material-UI
          </span>
        </div>
        <div
          class="MuiChip-root MuiChip-outlined MuiChip-sizeMedium MuiChip-colorPrimary MuiChip-outlinedPrimary home__skills-chip css-mo0pjq-MuiChip-root"
        >
          <span
            class="MuiChip-label MuiChip-labelMedium css-16cgrcw-MuiChip-label"
          >
            SCSS
          </span>
        </div>
        <div
          class="MuiChip-root MuiChip-outlined MuiChip-sizeMedium MuiChip-colorPrimary MuiChip-outlinedPrimary home__skills-chip css-mo0pjq-MuiChip-root"
        >
          <span
            class="MuiChip-label MuiChip-labelMedium css-16cgrcw-MuiChip-label"
          >
            Vite
          </span>
        </div>
        <div
          class="MuiChip-root MuiChip-outlined MuiChip-sizeMedium MuiChip-colorPrimary MuiChip-outlinedPrimary home__skills-chip css-mo0pjq-MuiChip-root"
        >
          <span
            class="MuiChip-label MuiChip-labelMedium css-16cgrcw-MuiChip-label"
          >
            NX
          </span>
        </div>
        <div
          class="MuiChip-root MuiChip-outlined MuiChip-sizeMedium MuiChip-colorPrimary MuiChip-outlinedPrimary home__skills-chip css-mo0pjq-MuiChip-root"
        >
          <span
            class="MuiChip-label MuiChip-labelMedium css-16cgrcw-MuiChip-label"
          >
            Git
          </span>
        </div>
        <div
          class="MuiChip-root MuiChip-outlined MuiChip-sizeMedium MuiChip-colorPrimary MuiChip-outlinedPrimary home__skills-chip css-mo0pjq-MuiChip-root"
        >
          <span
            class="MuiChip-label MuiChip-labelMedium css-16cgrcw-MuiChip-label"
          >
            Responsive Design
          </span>
        </div>
      </div>
    </div>
  </div>
  <div
    class="home__features MuiBox-root css-0"
  >
    <h2
      class="MuiTypography-root MuiTypography-h3 MuiTypography-gutterBottom home__features-title css-hedpeh-MuiTypography-root"
    >
      What I Bring to the Table
    </h2>
    <div
      class="MuiGrid-root MuiGrid-container MuiGrid-direction-xs-row MuiGrid-spacing-xs-4 css-kqg16c-MuiGrid-root"
    >
      <div
        class="MuiGrid-root MuiGrid-direction-xs-row css-17fpwt7-MuiGrid-root"
      >
        <div
          class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root home__feature-card css-r6ekfc-MuiPaper-root-MuiCard-root"
          style="--Paper-shadow: 0px 2px 1px -1px rgba(0,0,0,0.2),0px 1px 1px 0px rgba(0,0,0,0.14),0px 1px 3px 0px rgba(0,0,0,0.12);"
        >
          <div
            class="MuiCardContent-root home__feature-card-content css-1lt5qva-MuiCardContent-root"
          >
            <div
              class="home__feature-card-icon home__feature-card-icon--code MuiBox-root css-0"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium feature-icon-code css-1umw9bq-MuiSvgIcon-root"
                data-testid="CodeIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M9.4 16.6 4.8 12l4.6-4.6L8 6l-6 6 6 6zm5.2 0 4.6-4.6-4.6-4.6L16 6l6 6-6 6z"
                />
              </svg>
            </div>
            <h3
              class="MuiTypography-root MuiTypography-h5 MuiTypography-gutterBottom css-wkkoyu-MuiTypography-root"
            >
              Clean Code
            </h3>
            <p
              class="MuiTypography-root MuiTypography-body1 css-1k1cusp-MuiTypography-root"
            >
              Writing maintainable, scalable, and well-documented code following best practices.
            </p>
          </div>
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-direction-xs-row css-17fpwt7-MuiGrid-root"
      >
        <div
          class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root home__feature-card css-r6ekfc-MuiPaper-root-MuiCard-root"
          style="--Paper-shadow: 0px 2px 1px -1px rgba(0,0,0,0.2),0px 1px 1px 0px rgba(0,0,0,0.14),0px 1px 3px 0px rgba(0,0,0,0.12);"
        >
          <div
            class="MuiCardContent-root home__feature-card-content css-1lt5qva-MuiCardContent-root"
          >
            <div
              class="home__feature-card-icon home__feature-card-icon--code MuiBox-root css-0"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium feature-icon-design css-1umw9bq-MuiSvgIcon-root"
                data-testid="BrushIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M7 14c-1.66 0-3 1.34-3 3 0 1.31-1.16 2-2 2 .92 1.22 2.49 2 4 2 2.21 0 4-1.79 4-4 0-1.66-1.34-3-3-3m13.71-9.37-1.34-1.34a.996.996 0 0 0-1.41 0L9 12.25 11.75 15l8.96-8.96c.39-.39.39-1.02 0-1.41"
                />
              </svg>
            </div>
            <h3
              class="MuiTypography-root MuiTypography-h5 MuiTypography-gutterBottom css-wkkoyu-MuiTypography-root"
            >
              Modern Design
            </h3>
            <p
              class="MuiTypography-root MuiTypography-body1 css-1k1cusp-MuiTypography-root"
            >
              Creating beautiful, intuitive user interfaces with attention to detail and user experience.
            </p>
          </div>
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-direction-xs-row css-17fpwt7-MuiGrid-root"
      >
        <div
          class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root home__feature-card css-r6ekfc-MuiPaper-root-MuiCard-root"
          style="--Paper-shadow: 0px 2px 1px -1px rgba(0,0,0,0.2),0px 1px 1px 0px rgba(0,0,0,0.14),0px 1px 3px 0px rgba(0,0,0,0.12);"
        >
          <div
            class="MuiCardContent-root home__feature-card-content css-1lt5qva-MuiCardContent-root"
          >
            <div
              class="home__feature-card-icon home__feature-card-icon--code MuiBox-root css-0"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium feature-icon-speed css-1umw9bq-MuiSvgIcon-root"
                data-testid="SpeedIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="m20.38 8.57-1.23 1.85a8 8 0 0 1-.22 7.58H5.07A8 8 0 0 1 15.58 6.85l1.85-1.23A10 10 0 0 0 3.35 19a2 2 0 0 0 1.72 1h13.85a2 2 0 0 0 1.74-1 10 10 0 0 0-.27-10.44zm-9.79 6.84a2 2 0 0 0 2.83 0l5.66-8.49-8.49 5.66a2 2 0 0 0 0 2.83"
                />
              </svg>
            </div>
            <h3
              class="MuiTypography-root MuiTypography-h5 MuiTypography-gutterBottom css-wkkoyu-MuiTypography-root"
            >
              Performance
            </h3>
            <p
              class="MuiTypography-root MuiTypography-body1 css-1k1cusp-MuiTypography-root"
            >
              Optimizing applications for speed, accessibility, and cross-browser compatibility.
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    class="home__cta MuiBox-root css-0"
  >
    <h2
      class="MuiTypography-root MuiTypography-h4 MuiTypography-gutterBottom css-18p8awu-MuiTypography-root"
    >
      Ready to Start Your Next Project?
    </h2>
    <h6
      class="MuiTypography-root MuiTypography-h6 home__cta-subtitle css-b6y2se-MuiTypography-root"
    >
      Let's work together to bring your ideas to life
    </h6>
    <a
      class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeLarge MuiButton-containedSizeLarge MuiButton-colorPrimary MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeLarge MuiButton-containedSizeLarge MuiButton-colorPrimary home__cta-button css-dnximp-MuiButtonBase-root-MuiButton-root"
      href="/contact"
      tabindex="0"
    >
      Start a Conversation
    </a>
  </div>
</div>
`;
